### DISCUSSION TITLE
Need Help with SEO-Friendly 301 Redirects for Site Migration from WordPress to Framer

### DISCUSSION THREAD

**Discussion by Tibo8** [community/community#132778] [Programming Help]
### Select Topic Area

Question

### Body

Hello everyone,

I'm currently working on a project where I'm migrating a client's website from WordPress to Framer. The client has a Basic plan on Framer, which unfortunately does not support 301 redirects, and upgrading to the Pro plan is financially challenging for us at the moment.

The old site pages need to be permanently redirected to new pages to maintain SEO value and backlinks.
I have 2 questions:

Custom Code Solution: Is there any JavaScript or other code that I could insert into the new Framer pages to effectively perform 301 redirects and retain SEO value? I'm concerned about the impact on SEO if the redirects are not implemented correctly.

External Service: Does anyone know of a free external service that can handle 301 redirects in a way that preserves the SEO authority of the old pages? I've heard of some services that manage redirections but am unsure about their effectiveness for SEO.

Any help, suggestions, or advice would be greatly appreciated. I'm particularly interested in solutions that ensure the SEO and backlink authority from the old WordPress pages are correctly transferred to the new Framer pages.

Thank you in advance for your assistance

**Comment by SajanGhimire1** [comment]
@Tibo8 
you can use JavaScript to perform client-side redirects. However, be aware that these are generally not as SEO-friendly as server-side 301 redirects because search engines may not interpret them in the same way.
```
<script type="text/javascript">
  window.location.href = "https://www.newdomain.com/new-page";
</script>
```
-Utilize Cloudflare to configure 301 redirects with Page rules,check if your domain registrar offers URL redirection services to set up 301 redirects, maintaining SEO integrity.

**Reply by Tibo8** [reply]
Unfortunately, my domain registrar does not provide URL redirection services (only domain redirection). I've looked into Cloudflare, and it seems I need to point the DNS from my domain registrar to Cloudflare and then from Cloudflare to Framer. Am I right ?
Is it free and will the SEO of the specific pages be preserved?

**Reply by SajanGhimire1** [reply]
-Sign up for a Cloudflare account if you don't have one already.
-In Cloudflare, add your site, and Cloudflare will scan your domain's DNS records.
-Cloudflare will provide you with nameservers to configure at your domain registrar.
-Change your domain’s nameservers at your registrar to the nameservers provided by Cloudflare.
-Ensure that all DNS records (like A, CNAME, MX) are correctly set in Cloudflare to match the current settings you have (unless directed by Framer for specific changes). This ensures there's no downtime or loss of service for email or subdomains.

-Navigate to the "Page Rules" section within your Cloudflare dashboard.
-Create new rules where you specify the pattern of the URLs from your old WordPress site that need to be redirected.
-Specify the destination URLs on your new Framer site with a 301 redirect status. This tells search engines that the move is permanent.

**Reply by Tibo8** [reply]
![step 1](https://github.com/user-attachments/assets/1b407775-077e-4d82-9d78-0f5271ce7737)
![step 2](https://github.com/user-attachments/assets/14e1312f-ef1a-4ebd-8e40-e0ab13856c66)
![step 3](https://github.com/user-attachments/assets/41f36750-6eea-4277-a9ee-569f562a7e36)
![step 4](https://github.com/user-attachments/assets/b7a2ec41-72e9-4fcd-a302-fd10a992ad0d)

I followed all the steps you mentioned, but now I want to stop everything. Upgrading to the highest Framer plan will cost me money, and I don't want to risk anything with this client since it's my first project and I'm just a teenager. I'm worried about losing his SEO and potentially ruining everything. 

His website isn't working anymore, and I'm anxious about the time it may take to implement DNS with Cloudflare. I really don't want to take any risks. 

I can't change the nameserver on the OVH panel (my client's domain registrar), so how can I stop this? Sorry for bothering you. Will I lose SEO since the website isn't live right now?

**Reply by Tibo8** [reply]
Oh! The domain registrar worked, but I've dropped to #2 in SEO for one keyword (it was #1 before). I'll try Cloudflare later, not for my first client who's already doing well in SEO. Thanks 54J4N for your help

**Reply by Tibo8** [reply]
I need your help to disable everything on Cloudflare because the landing page URL isn’t working, while the other pages are. I’d rather not take any risks. Again, sorry for the inconvenience.

**Reply by SajanGhimire1** [reply]
Hi @Tibo8 Morning, 
Okey you can disable easily if you have issue then ,

#follow me :

--Go to the Overview tab of your site in Cloudflare.
--Scroll down to the Advanced Actions section.
--Click on "Pause Cloudflare on Site."
--Go to the DNS tab in Cloudflare.
--For each DNS record, click the orange cloud icon to turn it grey, which disables Cloudflare proxying.
--Navigate to the Page Rules section.
--Toggle off or delete any active Page Rules you have set up.
--Go to the SSL/TLS tab.
--Set the SSL/TLS encryption mode to "Off."
--Navigate to the Firewall Rules section.
--Create a new rule to bypass Cloudflare for your domain by setting the action to "Bypass."

>  _This will disable Cloudflare, making your site run directly from your host. It should fix the landing page issue but removes Cloudflare's protections and speed enhancements._

**Reply by SajanGhimire1** [reply]
@Tibo8 
Just to note, disabling Cloudflare means you'll lose its protection against downtime. If there are any issues in the future, feel free to inform me.

**Reply by Tibo8** [reply]
Thank you, this is completely working now! I successfully published it with Framer and connected it to my domain, but it seems to have lost some LTR Power and Trust.

I’m 100% sure the pages previously held at least 2 in power and 1 in trust.

Is this loss typical, or should I restart my entire netlinking campaign? Will I naturally recover my LTR?

The previous page and domain:
![image](https://github.com/user-attachments/assets/e8e6f14f-0776-4f9d-8da4-3cc1fdec21e8)

The new page:
![image](https://github.com/user-attachments/assets/1e7ff815-ed1f-4952-b954-3ffc54535c85)

Then I came back and the page fell to 0:
![image](https://github.com/user-attachments/assets/85d105a5-0325-4a97-9e48-bd804f290976)

**Reply by SajanGhimire1** [reply]
@Tibo8 
 Recovery of LTR  can take time.😔
 temporary loss in link trust after a site migration is common. You don't necessarily need to restart your entire netlinking campaign right away focus on maintaining good SEO practices & monitor your site's performance. Over time, as your new site gains authority and visibility, you should see recovery in link trust.

**Reply by alexadam292** [reply]
I’m thrilled to join this forum and connect with fellow enthusiasts. I’m Alex adam from [SK Print & Signage](https://skprint.com.au/), a premier provider of printing and signage solutions based in Perth. We specialize in a variety of services, including business card printing and vehicle magnets, designed to help you make a lasting impression.

**Comment by ShahinIsmayilov** [comment]
Response #1

**Comment by ShahinIsmayilov** [comment]
Response #2

**Comment by ShahinIsmayilov** [comment]
Response #3

**Comment by jac-collab** [comment]
For handling 301 redirects without upgrading to Framer’s Pro plan, here are a couple of solutions:
You can use JavaScript for redirects, but keep in mind this isn’t as SEO-friendly as server-side redirects. To implement it, you can add a JavaScript snippet to the <head> of your new Framer pages:

<script>
  if (window.location.pathname === '/old-page') {
    window.location.replace('https://yournewsite.com/new-page');
  }
</script>

This code checks if the current URL matches the old page and then redirects to the new page. Note that while this method will redirect users, search engines might not handle it as well as server-side 301 redirects, which could impact your [SEO](https://menuland.ph/kfc-menu-prices-in-philippines/).

**Comment by alexadam292** [comment]
As a design technologist in Atlanta, I’m passionate about front-end web development, with a focus on creating engaging and functional user experiences. My site, [WP Troubles](https://wptroubles.com/), serves as a platform where I share insights, tutorials, and blog about the latest trends, techniques, and challenges in web design and development. By providing valuable content, my goal is to help others navigate the complexities of front-end development and enhance their skills in building dynamic websites. Whether you're looking to stay updated with industry best practices or improve your web design capabilities, WP Troubles offers resources to support your journey.

**Comment by ibizoobysa** [comment]
Moving your website from WordPress to Framer can be tricky when it comes to keeping your SEO in good shape. Framer's Basic plan doesn't support the best kind of redirects (called 301 redirects), which are important for maintaining your site's rankings. You can use JavaScript to help with redirects, but it’s not the best option for SEO—it's like hiring an agent who doesn’t fully understand your needs. It works, but not perfectly.

A better option would be to use services like Cloudflare or Redirection .io, which are more reliable for redirects. Think of them like finding an agent who really takes the time to understand your goals. Just like in [real estate](https://www.era.com/), trust and relationships take time, and after a migration, it’s normal for your SEO to drop a little at first. But if you stick with good practices, your rankings should improve again over time.

If the technical stuff feels too complicated, you can hire a virtual assistant to help. It’s like bringing in a pro agent to handle the details so you don’t lose any SEO value. They’ll know how to set things up right and save you the hassle, just like a great agent would with your real estate needs.

**Reply by sacrio23** [reply]
> Moving your website from WordPress to Framer can be tricky when it comes to keeping your SEO in good shape. Framer's Basic plan doesn't support the best kind of redirects (called 301 redirects), which are important for maintaining your site's rankings. You can use JavaScript to help with redirects, but it’s not the best option for SEO—it's like hiring an agent who doesn’t fully understand your needs. It works, but not perfectly.
> 
> A better option would be to use services like Cloudflare or Redirection .io, which are more reliable for redirects. Think of them like finding an agent who really takes the time to understand your goals. Just like in [real estate](https://www.era.com/), trust and relationships take time, and after a migration, it’s normal for your SEO to drop a little at first. But if you stick with good practices, your rankings should improve again over time.
> 
> If the technical stuff feels too complicated, you can hire a virtual assistant to help. It’s like bringing in a pro agent to handle the details so you don’t lose any SEO value. They’ll know how to set things [new minecraft update](https://mineapkhub.com/) up right and save you the hassle, just like a great agent would with your real estate needs.

Migrating from WordPress to Framer can hurt SEO if not handled well, especially since Framer’s Basic plan doesn’t support 301 redirects. JavaScript redirects work but aren’t ideal. Better to use tools like Cloudflare or Redirection.io. For a smoother setup, consider hiring a virtual assistant with SEO know-how.

**Comment by Kay-Day** [comment]
Response #1

**Comment by Kay-Day** [comment]
Response #2

**Comment by seo-engine** [comment]
When migrating a website from WordPress to Framer, setting up SEO-friendly 301 redirects is crucial. A 301 redirect ensures that old URLs automatically point to the new ones, preserving search rankings and preventing broken links. The process involves mapping old URLs to new ones, configuring redirects in Framer, and testing thoroughly to avoid 404 errors.

For a smooth and SEO-safe migration, **Tech Savy Crew** specializes in handling 301 redirects and website migrations, ensuring traffic and rankings are protected throughout the process.

**Comment by thomasshellbytm-ux** [comment]
Hello Community 👐

For a WordPress to Framer  migration I would include the 301 redirect mapping in the migration plan not something to deal with after the new site is live. 

First I would. Export the existing WordPress URL's and find the pages that get organic traffic, backlinks or rankings. Then I would create a URL-to-URL mapping where every important old URL goes to the relevant new Framer URL. 

A few things I would focus on are: 

Make sure the redirects are as direct as possible of making redirect chains. Do not send every old page to the homepage unless there is really no good replacement. Change links to point direclty to the new URL's. 

Change tags, XML sitemap, robots.txt and indexability after the migration. Test URLs and make sure they give the right 301 status code. Watch Google Search Console and organic traffic after the migration for crawl errors and changes in rankings. 

I would also take a list of URL's before changing DNS because finding missing URL's after the migration can be much harder. 

If you are working with an SEO team Triple Minds can also be considered for handling the SEO part of the migration including URL mapping, redirects, technical checks and monitoring after the migration. 

The main thing is to make the migration as smooth as possible, for both users and search engines. 

Hope this will help you .
